packages/graph/hash_graph/doc/structural-queries.md

```diff
@@ -0,0 +1,233 @@
+# Structural Queries in the Graph API
+
+Structural queries are the main entry point to read data from the Graph. They allow querying the Graph for ontology types and entities. They are the primary way to read data from the Graph.
+
+A structural query will always return a subgraph consisting of vertices and edges determined by the query itself. The Graph API supports two types of queries: entity query and type queries. Depending on the type of query, the root vertices of the subgraph will be either entities or types.
+
+A query consists of two parts: a set of filters and the depth of how far the resulting subgraph will be resolved.
+
+```json5
+{
+  filter: {
+    // ...
+  },
+  graphResolveDepths: {
+    // ...
+  },
+}
+```
+
+## Filter
+
+The filters are used to determine which root vertices will be included in the subgraph. A filter is composed of a set of condition of parameters and paths. A path is a sequence of steps that describe a path through the Graph. Depending on the type of the query the path will either start at an entity or a type. A parameter is a user defined variable that can be used to parameterize the query. As conditions, currently, equality and inequality conditions are supported.
+
+A path is a sequence of identifiers to describe a path based on the query type. For example, each type has a `baseUri` component that can be used to identify the type. If a query is supposed to filter all entity types, which have the URI `https://example.com/foo/`, the filter would look like
+
+```json5
+{
+  filter: {
+    equal: [{ path: ["baseUri"] }, { parameter: "https://example.com/foo/" }],
+  },
+}
+```
+
+Currently, three different parameter types can be specified: booleans, numbers, and strings.
+
+### Composability
+
+Filters can be composed, so it's possible to match on more than one query. Currently, three different compositions are supported: `all`, `any` and `not`. The `all` composition will match if all conditions are true. The `any` composition will match if at least one condition is true. The `not` composition will match if the condition is false.
```

Comment:
nit:
Perhaps it would be useful to write that the `all` and `any` compositions are equivalent to having `&&` or `||` between the conditions? Don't think it's hard to understand this as-is, though